tools/create_fake_backend.py

```diff
@@ -0,0 +1,178 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+
+init_py = Template(
+    """${HEADER}
+\"\"\"Fake ${capital_backend_name} device (${no_qubits} qubits)\"\"\"
+
+from .fake_${backend_name} import Fake${capital_backend_name}
+"""
+)
+
+fake_be_py = Template(
+    """${HEADER}
+
+\"\"\"
+Fake ${capital_backend_name} device (${no_qubits} qubits).
+\"\"\"
+
+import os
+from qiskit.providers.fake_provider import fake_backend
+
+
+class Fake${capital_backend_name}(fake_backend.FakeBackendV2):
+    \"\"\"A fake ${no_qubits} qubit backend.\"\"\"
+
+    dirname = os.path.dirname(__file__)
+    conf_filename = "conf_${backend_name}.json"
+    props_filename = "props_${backend_name}.json"
+    defs_filename = "defs_${backend_name}.json"
+    backend_name = "fake_${backend_name}"
+
+"""
+)
+
+reno_template = Template(
+    """---
+features:
+  - |
+    The fake backend :class:`~Fake${capital_backend_name}` was added with the information
+    from IBM Quantum `${system_name}` system.
+"""
+)
+
+
+def _insert_line_in_section(
+    line_to_insert, section, line_in_section_starts_with, backend_dir, file_to_modify
+):
+    tmp_file = os.path.join(backend_dir, "_tmp_.py")
+    with open(file_to_modify, "r") as original, open(tmp_file, "a+") as destination:
+        previous_line = ""
+        in_section = False
+        for line in original.readlines():
+            if line.startswith(section):
+                in_section = True
+            elif (
+                in_section
+                and line.startswith(line_in_section_starts_with)
+                and previous_line < line_to_insert < line
+            ):
+                in_section = False
+                destination.write(line_to_insert)
+            destination.write(line)
+    os.replace(tmp_file, file_to_modify)
+
+
+def _main():
+    parser = argparse.ArgumentParser(description="Create fake backend")
+    parser.add_argument("--dir", "-d", type=str, default=DEFAULT_DIR)
+    parser.add_argument("backend_name", type=str, default=None)
+    parser.add_argument("system_name", type=str, default=None)
+    parser.add_argument("--project", type=str, default=None)
+    parser.add_argument("--hub", type=str, default=None)
+    parser.add_argument("--group", type=str, default=None)
+    args = parser.parse_args()
+
+    backend_dir = os.path.join(args.dir, args.backend_name)
+    if os.path.isdir(backend_dir):
+        print(f"ERROR: Directory {backend_dir} already exists")
+        sys.exit(1)
+
+    provider = IBMQ.load_account()
+    if args.hub or args.group or args.project:
+        provider = IBMQ.get_provider(hub=args.hub, group=args.group, project=args.project)
+    ibmq_backend = provider.get_backend(args.system_name)
+
+    vars_ = {
+        "HEADER": HEADER,
+        "backend_name": args.backend_name,
+        "capital_backend_name": args.backend_name.capitalize(),
```

Comment:
I think there was a discussion that we don't need to add `BackendV1` fake backend any more. Ideally I would like to have `FakeVigo` to be a backend that has the latest backend version, `FakeVigoV1` and `FakeVigoV2` to have explicit backend version name in the class and `FakeVigo` is simply `FakeVigoV2` and it will be `FakeVigoV3` if there is a new backend version. 

But at the moment we don't have this distinction. Fake backends without explicit version like `FakeVigo` are all V1 backends. So I would suggest to add `V2` for all the backends that are created using this script to avoid confusion. What do you think?